homeassistant/components/mqtt/__init__.py

```diff
@@ -313,11 +313,13 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     if (conf := await async_fetch_config(hass, entry)) is None:
         # Bail out
         return False
+    client: MQTT = hass.data.setdefault(DATA_MQTT, MQTT(hass, entry, conf))
```

Comment:
We can't use `setdefault` here, an `MQTT()` will always be created.

Reply:
Can we not reuse the client?

Reply:
If I replace the setdefault() here the test `test_init.py`->`test_no_birth_message` fails.

I doubt if this is an improvement. I'll turn it to draft for now.